gpytorch/kernels/product_spectral_gp_kernel.py

```diff
@@ -0,0 +1,51 @@
+import torch
+
+from gpytorch.kernels import ProductKernel
+from torch.nn import ModuleList
+
+from .spectral_gp_kernel import SpectralGPKernel
+
+class ProductSpectralGPKernel(ProductKernel):
```

Comment:
If the `SpectralGPKernel` itself correctly supports the `last_dim_is_batch` argument, we should be able to get away with something like a `ProductStructureKernel`. What we want to do is think of `n x d` data as actually `d x n x 1` batch data, compute a `d x n x n` kernel matrix and then product across the `d` "batch" dimension.

Since GP models run in batch mode, getting `SpectralGPKernel` to support batch inputs should be relatively straightforward, and this would make the multi-d version of this kernel basically `d` times faster.

Reply:
One question here: we are attempting to include support for both product spectral kernels with shared latent GPs and independent latent GPs. Will `ProductStructureKernel` be able to support both cases?